095-stretch-cluster.md

```diff
@@ -0,0 +1,201 @@
+
+### Prerequisites
+
+- **Multiple Kubernetes clusters**: Stretch Kafka clusters will require multiple Kubernetes clusters.
+Ideally, an odd number of clusters (at least three) will be needed to maintain Kafka controller quorum and tolerate a Kubernetes cluster outage.
+
+- **Low Latency**: Kafka clusters should be deployed in environments that allow low-latency communication between Kafka brokers and controllers.
+Stretch Kafka clusters should be deployed in environments such as data centers or availability zones within a single region, and not across distant regions where high latency could impair performance.
+
+- **KRaft**: As Kafka and Strimzi transition towards KRaft-based clusters, this proposal focuses exclusively on enabling stretch deployments for KRaft-based Kafka clusters.
+While Zookeeper-based deployments are still supported, they are outside the scope of this proposal.
+
+- **A supported cloud native network technology**: To enable networking between Kubernetes clusters currently requires an additional technology stack.
+The remainder of this proposal and the prototype detailed within assumes the use of [Submariner](https://submariner.io/), although there is potential for [Cilium](https://cilium.io/) support in a future revision.
+The [Istio](https://istio.io/) service mesh has some experimental support for [Kubernetes Multi-Cluster Services (MCS)](https://multicluster.sigs.k8s.io/concepts/multicluster-services-api/) that might also warrant further research.
+
+
+### Design
+
+#### Topology of a stretch cluster
+
+![Stretch cluster topology](./images/095-stretch-cluster-topology.png)
+
+The diagram illustrates a topology comprising of three Kubernetes clusters.
+
+One of these clusters is designated as the "Central cluster", while any additional clusters are considered "remote".
+The central cluster acts as the control plane where a user will create their Kafka cluster and Kafka node pool definitions (Custom Resources).
+
+A Kafka node pool definition can be configured to run on any of the Kubernetes clusters, including the central cluster.
+
+Operators deployed to remote clusters are only responsible for reconciling Strimzi pod set resources that are created remotely by the operator running in the central cluster.
+
+This approach will allow users to manage the definition of their stretch Kafka cluster in a single location.
+
+### Prototype
+
+A working prototype can be deployed based upon Submariner technology using the steps outlined in a [draft README](https://aswinayyolath.github.io/stretch-kafka-docs/) that is being iteratively revised.
+The following sections will describe the key aspects of the prototype highlighting areas that would benefit from community feedback and input.
+
+#### User configuration of the cross-cluster Kubernetes network
+
+A cross-cluster network must be configured manually as a pre-requisite using a supported network technology.
```

Comment:
cross-cluster Kubernetes network --> see previous comment

Reply:
The prototype is using Multi-Cluster Service features of Submariner, so I think we should stick with that. We'll need some extra words when we can add more about Cilium.